crates/anvil-polkadot/tests/it/snapshot.rs

```diff
@@ -22,37 +22,23 @@ use polkadot_sdk::pallet_revive::{
 use std::collections::HashSet;
 use subxt::utils::H256;
 
-async fn assert_block_number_is_best_and_finalized(
+async fn assert_block_number_is_best(
     node: &mut TestNode,
     n: u64,
     wait_for_block_provider: Option<Duration>,
 ) {
     assert_eq!(std::convert::Into::<u64>::into(node.best_block_number().await), n);
     if let Some(duration) = wait_for_block_provider {
         tokio::time::sleep(duration).await;
-        let best_block = unwrap_response::<Block>(
-            node.eth_rpc(EthRequest::EthGetBlockByNumber(
-                alloy_eips::BlockNumberOrTag::Latest,
-                false,
-            ))
-            .await
-            .unwrap(),
-        )
-        .unwrap();
-        let n_as_u256 = pallet_revive::U256::from(n);
-        assert_eq!(best_block.number, n_as_u256);
-
-        let finalized_block = unwrap_response::<Block>(
```

Comment:
it was a good idea to keep this finalized assertion. I assume it is failing without the sleep because seeing a best block does not neccessarily mean seeing it finalized.

I suggest to add a retry step here and do a small sleep if the initial check fails